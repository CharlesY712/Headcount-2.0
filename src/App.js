import React, { Component } from 'react';
import './App.css';
import CardContainer from './CardContainer';
import DistrictRepository from './helper';
import kinderData from './data/kindergartners_in_full_day_program.js';
// import PropTypes from 'prop-types';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      districtStats: null
    };
  }

  componentDidMount() {
    this.getData(kinderData);
  }

  getData(stats) {
    const districtData = new DistrictRepository(stats);
    this.setState({
      districtStats: districtData
    });
  }

  render() {
    return (
      <div>
        <h1>Welcome to Headcount</h1>
        {this.state.districtStats &&
        <CardContainer data={this.state.districtStats}/>
        }
      </div>
    );
  }
}

// App.propTypes = {
//   // addIdea: PropTypes.func
// }

export default App;
