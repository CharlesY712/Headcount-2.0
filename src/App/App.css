@import url('https://fonts.googleapis.com/css?family=Noto+Sans:400,700');

h1 {
  font-size: 100px;
  text-align: center;
  background-color: #6495ED;
  margin-top: 0;
  margin-bottom: 20px;
  font-family: 'Noto Sans', sans-serif;

}