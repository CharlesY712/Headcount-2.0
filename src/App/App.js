import React, { Component } from 'react';
import './App.css';
import Search from '../Search/Search';
import CardContainer from '../CardContainer/CardContainer';
import CompareContainer from '../CompareContainer/CompareContainer';
import DistrictRepository from '../helper';
import kinderData from '../data/kindergartners_in_full_day_program.js';
import PropTypes from 'prop-types';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      districtStats: null,
      compareArray: [],
      compareStats: {}
    };

    this.district = new DistrictRepository(kinderData);
  }

  componentDidMount() {
    this.getData();
  }

  getData() {
    this.setState({
      districtStats: this.district.findAllMatches()
    });
  }

  findMatch = (input) => {
    const matches = this.district.findAllMatches(input);
    this.setState({
      districtStats: matches
    });
  }

  displayComparedCards = (event) => {
    const location = event.target.closest('.card').firstChild.innerText;
    const selectedCard = this.district.findByName(location);
    const found = 
    this.state.compareArray.find(stat => stat.location === location);
    
    if (this.state.compareArray.length < 2) {
      this.setState({
        compareArray: [...this.state.compareArray, selectedCard]
      });
    }
    
    if (found) {
      const filtered = 
      this.state.compareArray.filter(stats =>  stats.location !== location);
      this.setState({
        compareArray: filtered
      });
    }

  }

  displayComparison = (name1, name2) => {
    const comparedStats = 
    this.district.compareDistrictAverages(name1.location, name2.location);
    return comparedStats;
  }

  render() {
    return (
      <div>
        <h1>Welcome to Headcount</h1>
        <Search findMatch={this.findMatch}/>
        {this.state.districtStats &&
        <div>
          <CompareContainer 
            compareArray={this.state.compareArray}
            displayComparison={this.displayComparison}
            displayComparedCards={this.displayComparedCards}
          />
          <CardContainer 
            stats={this.state.districtStats}
            displayComparedCards={this.displayComparedCards}
            compareArray={this.state.compareArray}
          />
        </div>
        }
      </div>
    );
  }
}

App.propTypes = {
  addIdea: PropTypes.func
};

export default App;
