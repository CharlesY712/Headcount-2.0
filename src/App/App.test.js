import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import App from './App';

describe('App', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
  });

  it('should match the snapshot', () => {
    const wrapper = shallow(<App />, { disableLifecycleMethods: true });
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a default state', () => {
    const wrapper = shallow(<App />, {disableLifecycleMethods: true});
    expect(wrapper.state('districtStats')).toEqual(null);
    expect(wrapper.state('compareArray')).toEqual([]);
    expect(wrapper.state('compareStats')).toEqual({});
  });

  it('should update state districtStats when getData is called', () => {
    const wrapper = shallow(<App />, { disableLifecycleMethods: true });
    wrapper.instance().getData();
    expect(wrapper.state('districtStats').length).toEqual(181);
  });

  it('should update state when findMatch is called', () => {
    const matches = 'colorado';
    wrapper.instance().findMatch(matches);
    expect(wrapper.state('districtStats').length).toEqual(2);
  });

  // it('should invoke findByName when displayComparedCards is invoked', () => {
  //   const mockFindByName = jest.fn();
  //   const mockEvent = { target: { closest: jest.fn().mockImplementation() } };
  //   wrapper.instance().displayComparedCards(mockEvent);
  //   expect(mockFindByName).toHaveBeenCalled();
  // });

  // it('should place card into compare array if length is less than two', () => {

  // });
});
