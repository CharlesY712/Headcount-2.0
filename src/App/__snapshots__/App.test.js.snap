// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="title"
>
          Welcome to Headcount
</h1>,
        <Search
          findMatch={[Function]}
/>,
        <div>
          <CompareContainer
                    compareArray={Array []}
                    displayComparedCards={[Function]}
                    displayComparison={[Function]}
          />
          <CardContainer
                    compareArray={Array []}
                    displayComparedCards={[Function]}
                    stats={Array []}
          />
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Welcome to Headcount",
          "className": "title",
        },
        "ref": null,
        "rendered": "Welcome to Headcount",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "findMatch": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <CompareContainer
              compareArray={Array []}
              displayComparedCards={[Function]}
              displayComparison={[Function]}
/>,
            <CardContainer
              compareArray={Array []}
              displayComparedCards={[Function]}
              stats={Array []}
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "compareArray": Array [],
              "displayComparedCards": [Function],
              "displayComparison": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "compareArray": Array [],
              "displayComparedCards": [Function],
              "stats": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="title"
>
            Welcome to Headcount
</h1>,
          <Search
            findMatch={[Function]}
/>,
          <div>
            <CompareContainer
                        compareArray={Array []}
                        displayComparedCards={[Function]}
                        displayComparison={[Function]}
            />
            <CardContainer
                        compareArray={Array []}
                        displayComparedCards={[Function]}
                        stats={Array []}
            />
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Welcome to Headcount",
            "className": "title",
          },
          "ref": null,
          "rendered": "Welcome to Headcount",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "findMatch": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <CompareContainer
                compareArray={Array []}
                displayComparedCards={[Function]}
                displayComparison={[Function]}
/>,
              <CardContainer
                compareArray={Array []}
                displayComparedCards={[Function]}
                stats={Array []}
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "compareArray": Array [],
                "displayComparedCards": [Function],
                "displayComparison": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "compareArray": Array [],
                "displayComparedCards": [Function],
                "stats": Array [],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
